internal/graph/storagewrapper.go

```diff
@@ -132,7 +132,7 @@ func (c *CachedDatastore) ReadStartingWithUser(
 		b.WriteString(fmt.Sprintf("/%s", strconv.FormatUint(hasher.Sum64(), 10)))
 	}
 
-	return c.newCachedIterator(ctx, store, iter, b.String(), storage.GetInvalidIteratorByUserObjectTypeCacheKeys(store, subjects, filter.ObjectType))
+	return c.newCachedIteratorByUserObjectType(ctx, store, iter, b.String(), subjects, filter.ObjectType)
```

Comment:
unrelated to this PR: how does line 119 work? what if `userFilter` has no relation but a wildcard?